### ISSUE TITLE
The chains of compond generated by 'Bio.PDB.get_structure().header' are all lowercase

### ISSUE THREAD

**Issue by wangjiangyuan** [biopython/biopython#4176]
### Setup

I am reporting a problem with Biopython version, Python version, and operating
system as follows:
python: 3.7
Bio: 1.73
```python
import sys; print(sys.version)
import platform; print(platform.python_implementation()); print(platform.platform())
import Bio; print(Bio.__version__)
```

(*Please copy and run the above in your Python, and copy-and-paste the output*)

### Expected behaviour
![image](https://user-images.githubusercontent.com/48577567/203003762-92dc2933-aaeb-4a89-9235-28c71b986ba6.png)
The pdb-id is 7DK4, and I generated the dict of chains of compound by Bio.PDB.get_structure().header. The dict showed that the all the chains are lowercase. It may be lead some trouble.
![image](https://user-images.githubusercontent.com/48577567/203005324-b719e01c-91b9-450d-9bc8-418784acd6e5.png)

(*Please fill this in*)

### Actual behaviour

(*Please fill this in, and provide any exception message in full*)

### Steps to reproduce

(*Please fill this in *)

**Comment by peterjc** [comment]
Are you really using Biopython 1.73 (from December 2018, nearly four years ago), please try upgrading and retest.

**Comment by peterjc** [comment]
Well, it was worth checking if this was a version issue.

I've now download the PDB format file from https://www.rcsb.org/structure/7DK4 and don't understand what you think is a problem, the header starts:

```
HEADER    VIRAL PROTEIN                           23-NOV-20   7DK4              
TITLE     S-2H2-F3A STRUCTURE, TWO RBDS ARE UP AND ONE RBD IS DOWN, EACH RBD    
TITLE    2 BINDS WITH A 2H2 FAB.                                                
COMPND    MOL_ID: 1;                                                            
COMPND   2 MOLECULE: THE HEAVY CHAIN OF 2H2 FAB;                                
COMPND   3 CHAIN: a, c, e;                                                      
COMPND   4 MOL_ID: 2;                                                           
COMPND   5 MOLECULE: THE LIGHT CHAIN OF 2H2 FAB;                                
COMPND   6 CHAIN: b, d, f;                                                      
COMPND   7 MOL_ID: 3;                                                           
COMPND   8 MOLECULE: SPIKE GLYCOPROTEIN;                                        
COMPND   9 CHAIN: A, B, C;                                                      
COMPND  10 SYNONYM: S GLYCOPROTEIN,E2,PEPLOMER PROTEIN,SARS-COV-2 S PROTEIN;    
COMPND  11 ENGINEERED: YES                                                      
SOURCE    MOL_ID: 1;                                                            
SOURCE   2 ORGANISM_SCIENTIFIC: MUS MUSCULUS;                                   
...
```

i.e. The ``CHAIN`` entries are lower case in ``MOL_ID`` 1 and 2 (the heavy and light chains), but upper case in ``MOL_ID`` 3.

This looks like an oddity in the PDB file, any Biopython is simply parsing the header as it is. That's all ``.header`` is from the structure object.

**Comment by wangjiangyuan** [comment]
The entries are lower case in 1 and 2 (the heavy and light chains), but upper case in 3. However, Biopython give me the incorrect header, the result shows the entries are lower case in 1,2 and 3, which should be upper case.

>>>Bio.__version__
'1.80'
>>>s=PDBParser()
>>>q=s.get_structure('7dk4','7dk4.pdb')
>>>q.header['compound']
{'1': {'misc': '', 'molecule': 'the heavy chain of 2h2 fab', 'chain': 'a, c, e'}, '2': {'misc': '', 'molecule': 'the light chain of 2h2 fab', 'chain': 'b, d, f'}, '3': {'misc': '', 'molecule': 'spike glycoprotein', 'chain': 'a, b, c', 'synonym': 's glycoprotein,e2,peplomer protein,sars-cov-2 s protein', 'engineered': 'yes'}}

**Comment by peterjc** [comment]
Ah, that's clearer. The screenshot was a little hard to read.

**Comment by peterjc** [comment]
It looks like *everything* in the header has been converted to lower case (and almost everything was in upper case to start with).

**Comment by wangjiangyuan** [comment]
Yes, it can cause some trouble. I tried to fix it but I failed. Can you give me a hand?

**Comment by peterjc** [comment]
Looks like this line: https://github.com/biopython/biopython/blob/biopython-180/Bio/PDB/parse_pdb_header.py#L227

Of course, things might break if you remove the ``.lower()``

**Comment by wangjiangyuan** [comment]
Yeah, I tried to remove the .lower() and I failed because I got this result. The other lower case chains are missing. I think it's not simple to fix it.
Bio.version
'1.80'
s=PDBParser()
q=s.get_structure('7dk4','7dk4.pdb')
q.header['compound']

{'1': {'misc': '', 'MOL_ID': '3', 'MOLECULE': 'SPIKE GLYCOPROTEIN', 'CHAIN': 'A, B, C', 'SYNONYM': 'S GLYCOPROTEIN,E2,PEPLOMER PROTEIN,SARS-COV-2 S PROTEIN', 'ENGINEERED': 'YES'}}

**Comment by rob-miller** [comment]
Probably the lower() is there just to make the header output more pleasant
for human reading.  Historically (like the 2004 date at the top of
parse_pdb_header.py) the PDB header information was not as reliably
parseable over the entire database as it appears from an individual entry
or even the documentation.

Looking at parse_pdb_header.py (thanks Peter for finding the line), it's
not doing any parsing of the COMPND line other than looking for an E.C.
number, so doesn't know anything about the chain spec given here in 7dk4.
The chain IDs in Biopython are collected from the ATOM records I believe,
and for 7dk4 the entries are correct for the SMCRA architecture as
described in http://biopython.org/DIST/docs/tutorial/Tutorial.html#sec193 :

7dk4 1 ('7dk4',):('7dk4',) ['a', 'b', 'A', 'B', 'C', 'c', 'd', 'e', 'f'] --
4470 of 4470 residue IDs match; 4470 residues 34707 atom coords, 34707 full
IDs of 34707 atoms (0 disordered) match : ALL OK

(Output from my own code above).  So the short form is that very little is
reliably parsed from PDB header data by Biopython and is primarily just
there for human (not machine) annotation, and you should use the SMCRA data
structure to extract this information.

**Comment by wangjiangyuan** [comment]
It is useful to get all chains with correct IDs. However, it can not show what are compound the chains belong to. It may provide less information for the chains rather than the 'get_structure('7dk4','7dk4.pdb').header'. And now I need this information that compound and their chains.
7dk4 1 ('7dk4',):('7dk4',) ['a', 'b', 'A', 'B', 'C', 'c', 'd', 'e', 'f']

**Comment by wangjiangyuan** [comment]
I have fixed it. 
https://github.com/biopython/biopython/blob/biopython-180/Bio/PDB/parse_pdb_header.py#L227  remove '.lower()'
https://github.com/biopython/biopython/blob/biopython-180/Bio/PDB/parse_pdb_header.py#L237  "mol_id" change 'MOL_ID'
And I got the correct result.

{'compound': {'1': {'misc': '', 'MOLECULE': 'THE HEAVY CHAIN OF 2H2 FAB', 'CHAIN': 'a, c, e'}, '2': {'misc': '', 'MOLECULE': 'THE LIGHT CHAIN OF 2H2 FAB', 'CHAIN': 'b, d, f'}, '3': {'misc': '', 'MOLECULE': 'SPIKE GLYCOPROTEIN', 'CHAIN': 'A, B, C', 'SYNONYM': 'S GLYCOPROTEIN,E2,PEPLOMER PROTEIN,SARS-COV-2 S PROTEIN', 'ENGINEERED': 'YES'}}}

**Comment by rob-miller** [comment]
The correct fix if you'd like to contribute would be to add lines to
correctly parse the COMPND record according to the format described here:

https://www.wwpdb.org/documentation/file-format-content/format33/sect2.html#COMPND
